ct_data_schema.json

```
@@ -0,0 +1,134 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "NASA Ti64 30um Layers Data",
+    "description": "Metadata associated with experimental results from NASA Ti64 30um layers study",
+    "type": "object",
+    "properties": {
+      "Parameter_Label": {
+        "title": "Parameter Label",
+        "description": "Must start with A, B, or C followed by two digits.",
+        "type": "string",
+        "pattern": "^[A-C][0-9]{2}$",
+        "propertyOrder": 0
+      },
+      "DOE_Code": {
+        "title": "DOE Code",
+        "description": "Derived from the first character of Parameter_Label (A→1, B→2, C→3).",
+        "type": "integer",
+        "readOnly": true,
+        "watch": {
+          "Parameter_Label": "root.Parameter_Label"
+        },
+        "propertyOrder": 1
+      },
+      "DOE_ID": {
+        "title": "DOE ID",
+        "description": "Derived from the Parameter_Label",
+        "type": "string",
+        "format": "custom",
+        "template": "{{Parameter_Label}}",
```

It's not using any of your helpers and simply copying the value of `Parameter_Label`. Shouldn't it be:
```suggestion
        "template": "{{firstChar Parameter_Label}}",
```
?

---

@Xarthisius, I have added corrected code. Please review the changes. Thanks!